lib/concurrent/executor/ruby_thread_pool_executor.rb

```diff
@@ -292,11 +295,14 @@ def ns_reset_if_forked
     class Worker
       include Concern::Logging
 
-      def initialize(pool)
+      def initialize(pool, id: nil)
         # instance variables accessed only under pool's lock so no need to sync here again
         @queue  = Queue.new
         @pool   = pool
         @thread = create_worker @queue, pool, pool.idletime
+        if Concurrent.on_cruby? && Concurrent.ruby_version(:>=, 2, 3, 0)
```

Comment:
A feature check sounds better here, and would then work on other Ruby implementations too:
```ruby
if @thread.respond_to?(:name=)
  ...
```

Reply:
done